autoscale/src/main/java/com/sequenceiq/periscope/service/YarnBasedScalingAdjustmentService.java

```diff
@@ -82,28 +67,29 @@ public void pollYarnMetricsAndScaleCluster(Cluster cluster, String pollingUserCr
 
         LOGGER.info("Various counts: hostFqdnsToInstanceId={}, servicesHealthyHostFqdnsToInstanceId: {}, stoppedHostFqdnsToInstanceId: {}, " +
                         "maxResourceValueOffset={}, minResourceValueOffset={}, maxAllowedUpScale={}, maxAllowedDownScale={}",
-                existingHostGroupSize, serviceHealthyHostGroupSize, stoppedHostInstanceIds.size(),
+                existingHostGroupSize, serviceHealthyHostGroupSize, policyHostGroupInstanceInfo.getStoppedHostInstanceIds().size(),
                 maxResourceValueOffset, minResourceValueOffset, maxAllowedUpScale, maxAllowedDownScale);
 
         Optional<Integer> maxDecommissionNodeCount = Optional.of(maxResourceValueOffset)
                 .filter(mandatoryDownscale -> mandatoryDownscale < 0).map(downscale -> -1 * downscale);
 
         YarnScalingServiceV1Response yarnResponse =
-                yarnMetricsClient.getYarnMetricsForCluster(cluster, stackV4Response, policyHostGroup, pollingUserCrn, maxDecommissionNodeCount);
+                yarnMetricsClient.getYarnMetricsForCluster(cluster, stackV4Response, policyHostGroupInstanceInfo.getPolicyHostGroup(),
+                        pollingUserCrn, maxDecommissionNodeCount);
 
         if (cluster.getUpdateFailedDetails() != null && pollingUserCrn.equals(cluster.getMachineUserCrn())) {
             // Successful YARN API call
             clusterService.setUpdateFailedDetails(cluster.getId(), null);
```

Comment:
I know it is not touched now, but can we have logs here what was the details of the failure?
Why does it call setUpdateFailedDetails with `null` details when based on the condition there should be details?

Reply:
Added a log line, this block is meant to reset updateFailedDetails in case of a successful API call made wth machineUserCrn.

Reply:
Thanks, now it is more clear.